src/uu/tail/src/args.rs

```diff
@@ -358,8 +408,32 @@ fn parse_num(src: &str) -> Result<Signum, ParseSizeError> {
 }
 
 pub fn parse_args(args: impl uucore::Args) -> UResult<Settings> {
-    let matches = uu_app().try_get_matches_from(arg_iterate(args)?)?;
-    Settings::from(&matches)
+    let args_vec: Vec<OsString> = args.collect();
+    let clap_args = uu_app().try_get_matches_from(args_vec.clone());
+    let clap_result = match clap_args {
+        Ok(matches) => Ok(Settings::from(&matches)?),
+        Err(err) => Err(err.into()),
+    };
+
+    // clap misinterprets obsolete args as first input (starting with +) or Err
```

Comment:
I think the logic is good, but let's expand the explanation a bit. Something along these lines (feel free to adapt or expand further):

> There are a few cases of ambiguity between clap parsing and the obsolete syntax:
>  1. Cases like `tail -f foo` and `tail -c 5`, where the first argument is both a valid flag and a valid first argument of the obsolete syntax. These should be handled as normal clap arguments.
> 2.  Cases like `tail +f bar` where `+f` is parsed as a filename by clap. In this case, clap succeeds, but we still want to discard the result in favor of obsolete parsing. So below, we check whether the first argument starts with `+` even when clap succeeds.